docs/dev_guide/migration/debug-with-codepin.rst

```diff
@@ -0,0 +1,224 @@
+            VAR_SCHEMA_0, (long *)&d_a, VAR_SCHEMA_1, (long *)&d_result);
+
+        q_ct1.memcpy(h_result, d_result, vectorSize * sizeof(sycl::int3)).wait();
+
+        // Print the result
+        for (int i = 0; i < vectorSize; ++i) {
+            std::cout << "Result[" << i << "]: (" << h_result[i].x() << ", "
+                    << h_result[i].y() << ", " << h_result[i].z() << ")\n";
+        }
+    }
+
+    /*
+    Execution Result:
+    Result[0]: (2, 3, 4)
+    Result[1]: (2, 3, 4)
+    Result[2]: (2, 3, 4)
+    Result[3]: (1, 1, 1) <--- incorrect result
+    */
+
+``dpct_output_debug/example.cu`` is the instrumented CUDA program:
+
+.. code-block:: c++
+
+    //dpct_output_debug/example.cu
+    #include "generated_schema.hpp"
+    #include <dpct/codepin/codepin.hpp>
+    #include <iostream>
+
+    __global__ void vectorAdd(int3 *a, int3 *result) {
+        int tid = blockIdx.x * blockDim.x + threadIdx.x;
+        result[tid].x = a[tid].x + 1;
+        result[tid].y = a[tid].y + 1;
+        result[tid].z = a[tid].z + 1;
+    }
+
+    int main() {
+        const int vectorSize = 4;
+        int3 h_a[vectorSize], h_result[vectorSize];
+        int3 *d_a, *d_result;
+        for (int i = 0; i < vectorSize; ++i)
+            h_a[i] = make_int3(1, 2, 3);
+
+        cudaMalloc((void **)&d_a, vectorSize * sizeof(int3));
+        dpct::experimental::get_ptr_size_map()[*((void **)&d_a)] =
+            vectorSize * sizeof(int3);
+        cudaMalloc((void **)&d_result, vectorSize * sizeof(int3));
+        dpct::experimental::get_ptr_size_map()[*((void **)&d_result)] =
+            vectorSize * sizeof(int3);
+
+        // Copy host vectors to device
+        cudaMemcpy(d_a, h_a, vectorSize * 12, cudaMemcpyHostToDevice);
+
+        // Launch the CUDA kernel
+        dpct::experimental::gen_prolog_API_CP(
+            "example.cu:38:3", 0, VAR_SCHEMA_0,
+            (long *)&d_a, VAR_SCHEMA_1, (long *)&d_result);
+        vectorAdd<<<1, 4>>>(d_a, d_result);
+
+        // Copy result from device to host
+        dpct::experimental::gen_epilog_API_CP(
+            "example.cu:38:3", 0, VAR_SCHEMA_0,
+            (long *)&d_a, VAR_SCHEMA_1, (long *)&d_result);
+        cudaMemcpy(h_result, d_result, vectorSize * sizeof(int3),
+                    cudaMemcpyDeviceToHost);
+
+        // Print the result
+        for (int i = 0; i < vectorSize; ++i) {
+            std::cout << "Result[" << i << "]: (" << h_result[i].x << ", "
+                    << h_result[i].y << ", " << h_result[i].z << ")\n";
+        }
+    }
+
+    /*
+    Execution Result:
+    Result[0]: (2, 3, 4)
+    Result[1]: (2, 3, 4)
+    Result[2]: (2, 3, 4)
+    Result[3]: (2, 3, 4)
+    */
+
+After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_debug/example.cu``, following report will be generated.
+
+.. figure:: /_images/codepin_example_report.png
+
+The report helps the user to identify where the runtime behaivor of the CUDA and the SYCL version start diverge.
```

Comment:
Change to "where the runtime behavior of the CUDA and the SYCL version start to diverge from one another."